GoogleContainerTools/skaffold
# fix deployment stabilize max deadline logic to ignore if d.Spec.Deadline is kubernetes default.

**Fixes**:  #3772

Tests performed:
1. Build the binary
```
make
cd examples/microservices/
```

2.  Make a change in any on the yamls.
```
e.g.
tejaldesai@tejaldesai-macbookpro2 microservices (fix_default)tail leeroy-app/kubernetes/deployment.yaml 
  template:
    metadata:
      labels:
        app: leeroy-app
    spec:
      containers:
      - name: leeroy-app
        image: leeroy-app1  --------->>> Invalid image name
        ports:
        - containerPort: 50051
```
3. Make sure deployments errors out within 2 mins.
```
tejaldesai@tejaldesai-macbookpro2 microservices (fix_default)
../../out/skaffold run --status-check -d gcr.io/tejal-test
```
